### PR TITLE
Add call to the OpenMC Depletion Solver

### PR DESCRIPTION
**Pull request by abachma2**
This PR adds new functionality to the DepleteReactor archetype:
- Call the OpenMC depletion sovler during the `transmute` method
- Write the resulting composition to an xml file 
- Add state variables for the path to the directory with OpenMC model files and the name of the decay chain file 

The CI fails because importing openmc into the archetype (i.e. ``import openmc``) causes a segmentation fault. The seg fault is after the simulation runs and writes the results to the output file and is trying to close python imports (I think). So it all still mostly runs, and I have manually compared the results of the output database to the expected values in the test suite. An issue (#12) has been created to help resolve the seg fault.

### CODE REVIEW

File: .github/workflows/test-openmcyclus.yml
Comment (nsryan2) [inline .github/workflows/test-openmcyclus.yml:61]: What does this mean?

File: README.md
Comment (nsryan2) [inline README.md:7]:
```suggestion
Ensure you install Cyclus and OpenMC, and their dependencies, before installing OpenMCyclus.
```

File: README.md
Comment (nsryan2) [inline README.md:42]:
```suggestion
## Running
```
Should this be a separate subsection from installation?

File: README.md
Comment (nsryan2) [inline README.md:41]:
```suggestion

### Testing
In progress.
```
Should there be a testing subsubsection?

File: README.md
Comment (nsryan2) [inline README.md:15]: Would it make sense to put this whole sequence into a code block?

File: README.md
Comment (nsryan2) [reply README.md:7]: In the writing checklist, we say never to have empty sections.

File: openmcyclus/DepleteReactor.py
Comment (nsryan2) [inline openmcyclus/DepleteReactor.py:12]:
```suggestion
```
Wasn't used

File: openmcyclus/DepleteReactor.py
Comment (nsryan2) [inline openmcyclus/DepleteReactor.py:13]:
```suggestion
```
Wasn't used.

File: tests/integration_tests/test_depletereactor.py
Comment (nsryan2) [inline tests/integration_tests/test_depletereactor.py:126]:
```suggestion
        assert all(quantities == [10]*len(tbl))
```
Would it reduce the chance for errors if you add times later if you changed it to this? For test_resources too?

**Review by nsryan2 — COMMENTED** [review]
Mostly LGTM, I saw Sun Myung's comment so I won't approve quite yet, I had some minor README comments, package comments, and some stuff on the tests.

File: openmcyclus/DepleteReactor.py
Comment (smpark7) [inline openmcyclus/DepleteReactor.py:488]: Are the extra set of parentheses required?

File: .github/workflows/test-openmcyclus.yml
Comment (abachma2) [reply .github/workflows/test-openmcyclus.yml:47]: It looks like right now there is a syntax error in the ci scripts, so I will fix that. But because of a seg fault when importing OpenMC into a cyclus script (See issue #12), there isn't much point in running the integration tests in a CI environment, because they will all fail due to the seg fault. I can make sure that the environment builds and the unit tests pass, but that's the best I can do until Issue #12 is resolved.

File: .github/workflows/test-openmcyclus.yml
Comment (abachma2) [reply .github/workflows/test-openmcyclus.yml:61]: I'm not sure how that ended up there. I must have accidentally copied whatever was in my clipboard into the file. It's been removed!

**Review by smpark7 — CHANGES_REQUESTED** [review]
I didn't run anything but I looked through your code. Here are my review comments.

File: .github/workflows/test-openmcyclus.yml
Comment (smpark7) [reply .github/workflows/test-openmcyclus.yml:47]: Oh yes sorry for not reading your PR comments properly

File: README.md
Comment (abachma2) [reply README.md:7]: Those lines of code weren't supposed to be deleted. Weird.

File: README.md
Comment (abachma2) [reply README.md:42]: No. Running the code/calling the archetype in a cyclus input is different than installing it.

File: .github/workflows/test-openmcyclus.yml
Comment (abachma2) [reply .github/workflows/test-openmcyclus.yml:47]: All good. It's important to have the CI as functional as possible. I'll fix the syntax error int he CI on this branch, but do the bulk of the CI changes/maintenance on another branch to keep this PR from getting too big.

File: .github/workflows/test-openmcyclus.yml
Comment (abachma2) [reply .github/workflows/test-openmcyclus.yml]: No. I'm not sure how it got in there, but it has been removed.

File: README.md
Comment (abachma2) [reply README.md:99]: I think so. Any open tag has to be closed with the same keyword, as far as I've seen. I see that I have missed one or two. Will fix!

File: examples/complex.xml
Comment (abachma2) [reply examples/complex.xml:108]: Maybe? I'll play around with it. Can we make this an issue so that this doesn't hold up the merging of this PR?

File: openmc-xs.bash
Comment (abachma2) [reply openmc-xs.bash:5]: You don't need an ANL account to access the files. I got this script from [saltproc's CI](https://github.com/arfc/saltproc/blob/master/scripts/ci/openmc-xs.bash). I don't think this script is used right now, but was used previously. I will remove this file if still unused after figuring out the CI environment.

File: openmcyclus/DepleteReactor.py
Comment (abachma2) [reply openmcyclus/DepleteReactor.py:182]: spaces are not required, just the commas. The print statements are for diagnostics during development and will be removed for a complete version of the code.

File: openmcyclus/DepleteReactor.py
Comment (abachma2) [reply openmcyclus/DepleteReactor.py:374]: It is extra. Good catch.

File: openmcyclus/DepleteReactor.py
Comment (abachma2) [reply openmcyclus/DepleteReactor.py:488]: No, it just makes it easier to read for me.

File: openmcyclus/DepleteReactor.py
Comment (abachma2) [reply openmcyclus/DepleteReactor.py:371]: In what way? How would you suggest updating the index names?

File: README.md
Comment (smpark7) [reply README.md]: Btw I think there are five `val` and one `assem_size` tags which are missing the slash key

File: examples/complex.xml
Comment (smpark7) [reply examples/complex.xml:108]: Sure that's okay

File: openmcyclus/DepleteReactor.py
Comment (smpark7) [reply openmcyclus/DepleteReactor.py:369]: Renaming `jj` and `kk` as `ii` and `jj`. But leaving it as is is fine too

File: README.md
Comment (abachma2) [reply README.md:50]: All it would contain would be the sentence about running `pytest`.

File: tests/integration_tests/test_depletereactor.py
Comment (abachma2) [reply tests/integration_tests/test_depletereactor.py:126]: This part of the code is testing what's in the output database. So your code suggestion is just testing if all of the entries in the table are of value 10, not specifically if there are 9 occurrences of value 10 in the table. I suppose that by testing for `[10]*9` that's also checking that there are 9 values in the table, same as l.124. but I'm okay with being redundant in my testing.

File: README.md
Comment (abachma2) [reply README.md]: Yep. It should all be fixed now.

**Review by smpark7 — APPROVED** [review]
LGTM! @nsryan2 go ahead and merge if you approve too.

**Pull request merged**
